mlir/lib/Dialect/Rock/IR/RockDialect.cpp

```diff
@@ -1514,6 +1517,54 @@ LogicalResult AccelGemmOp::verify() {
     return emitOpError("K dimensions don't match");
   return success();
 }
+
+//===----------------------------------------------------------------------===//
+// GridwiseAttentionAccelOp
+//===----------------------------------------------------------------------===//
+LogicalResult GridwiseAttentionAccelOp::verify() {
+  RockAccelTuningParamAttrInterface gemm0TuningParams = getParams();
+  int64_t gemm0kpack = gemm0TuningParams.getKpack();
+  int64_t gemm0KpacksPerBlock = gemm0TuningParams.getKpackPerBlock();
+  int64_t gemm0MPerBlock = gemm0TuningParams.getMPerBlock();
+  int64_t gemm0NPerBlock = gemm0TuningParams.getNPerBlock();
+  int64_t gemm0KPerBlock = gemm0kpack * gemm0KpacksPerBlock;
+  if (gemm0NPerBlock % gemm0kpack != 0) {
+    return emitError("NPerBlock should be divisble by kpack.");
+  }
+
+  // Calculate LDS requirement
+  MemRefType typeQ = getQueries().getType();
+  Type elemTypeQ = typeQ.getElementType();
+  MemRefType typeK = getKeys().getType();
+  Type elemTypeK = typeK.getElementType();
+  MemRefType typeV = getValues().getType();
+  Type elemTypeV = typeV.getElementType();
+  MemRefType typeO = getOut().getType();
+  ArrayRef<int64_t> outShape = typeO.getShape();
+  int64_t gemm1N = outShape[2];
+
+  // TODO: we can definitely improve re-use of LDS buffers
+  // through further refactors to blockwise gemm to accept
+  // LDS buffers that are larger than required -- Hence
+  // we can overlap following buffers between gemms.
+  int64_t gemm0ALdsSizeBytes = (gemm0KPerBlock * gemm0MPerBlock) *
+                               (elemTypeQ.getIntOrFloatBitWidth() / 8);
+  int64_t gemm0BLdsSizeBytes = (gemm0KPerBlock * gemm0NPerBlock) *
+                               (elemTypeK.getIntOrFloatBitWidth() / 8);
+  // Current implementation does the second gemm using the type of V input.
+  int64_t gemm1ALdsSizeBytes = (gemm0NPerBlock * gemm0MPerBlock) *
+                               (elemTypeV.getIntOrFloatBitWidth() / 8);
+  int64_t gemm1BLdsSizeBytes =
+      (gemm0NPerBlock * gemm1N) * (elemTypeV.getIntOrFloatBitWidth() / 8);
+  int64_t totalLDSSize = gemm0ALdsSizeBytes + gemm0BLdsSizeBytes +
+                         gemm1ALdsSizeBytes + gemm1BLdsSizeBytes;
+  if (totalLDSSize > 64 * 1024) {
```

Comment:
This'll be an architecture-specific constant Eventually, worth leaving a note here. (It also might alreaby be anch-specific but we've been being conservative about it)

Reply:
do you want me to add a comment here ? 
Agreed that it needs to go to a lookup but in a future PR that would remove such checks for just gemm as well.

Reply:
I'll make a ticket about it